io/net/sriov_device_test.py

```diff
@@ -0,0 +1,235 @@
+#!/usr/bin/python
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2018 IBM
+# Author: Bismruti Bidhibrata Pattjoshi <bbidhibr@in.ibm.com>
+# Authors: Abdul haleem <abdhalee@linux.vnet.ibm.com>
+
+'''
+Tests for Sriov logical device
+'''
+
+import netifaces
+try:
+    import pxssh
+except ImportError:
+    from pexpect import pxssh
+from avocado import Test
+from avocado import main
+from avocado.utils import process
+from avocado.utils.software_manager import SoftwareManager
+from avocado.utils.network.interfaces import NetworkInterface
+from avocado.utils.network.hosts import LocalHost
+
+
+class CommandFailed(Exception):
+
+    '''
+    Defines the exception called when a
+    command fails
+    '''
+
+    def __init__(self, command, output, exitcode):
+        Exception.__init__(self, command, output, exitcode)
+        self.command = command
+        self.output = output
+        self.exitcode = exitcode
+
+    def __str__(self):
+        return "Command '%s' exited with %d.\nOutput:\n%s" \
+               % (self.command, self.exitcode, self.output)
+
+
+class NetworkSriovDevice(Test):
+    '''
+    adding and deleting logical sriov device through
+    HMC.
+    '''
+    def setUp(self):
+        '''
+        set up required packages and gather necessary test inputs
+        '''
+        smm = SoftwareManager()
+        packages = ['rsct.basic', 'rsct.core.utils',
+                    'rsct.core', 'DynamicRM']
+        for pkg in packages:
+            if not smm.check_installed(pkg) and not smm.install(pkg):
+                self.cancel('%s is needed for the test to be run' % pkg)
+        self.hmc_ip = self.get_mcp_component("HMCIPAddr")
+        if not self.hmc_ip:
+            self.cancel("HMC IP not got")
+        self.hmc_pwd = self.params.get("hmc_pwd", '*', default=None)
+        self.hmc_username = self.params.get("hmc_username", '*', default=None)
+        self.lpar = self.get_partition_name("Partition Name")
+        if not self.lpar:
+            self.cancel("LPAR Name not got from lparstat command")
+        self.login(self.hmc_ip, self.hmc_username, self.hmc_pwd)
+        cmd = 'lssyscfg -r sys  -F name'
+        output = self.run_command(cmd)
+        self.server = ''
+        for line in output.splitlines():
+            if line in self.lpar:
+                self.server = line
+                break
+        if not self.server:
+            self.cancel("Managed System not got")
+        self.sriov_adapter = self.params.get('sriov_adapter', '*', default=None)
```

Comment:
can we have multiple adapters to create more than one sriov device ?

Reply:
make sure multiple sriov_adapter input is space separated

we need multiple sriov in cases where bonding is tested, so sriov device coming from 2 different adapter port